contrib/src/templates/mod.rs

```diff
@@ -214,27 +242,46 @@ impl Template {
     ///
     /// Returns `Some` if the template could be rendered. Otherwise, returns
     /// `None`. If rendering fails, error output is printed to the console.
+    /// `None` is also returned if a Template fairing has not been attached.
     ///
     /// # Example
     ///
     /// ```rust
     /// use std::collections::HashMap;
+    ///
+    /// extern crate rocket;
+    /// extern crate rocket_contrib;
+    ///
     /// use rocket_contrib::Template;
     ///
-    /// // Create a `context`. Here, just an empty `HashMap`.
-    /// let mut context = HashMap::new();
+    /// fn main() {
+    ///     let client =
+    ///         rocket::local::Client::new(rocket::ignite()
+    ///             .attach(Template::fairing())
+    ///         ).expect("valid rocket");
     ///
-    /// # context.insert("test", "test");
-    /// # #[allow(unused_variables)]
-    /// let template = Template::show("templates/", "index", context);
+    ///     // Create a `context`. Here, just an empty `HashMap`.
+    ///     let mut context = HashMap::new();
+    ///
+    ///     # context.insert("test", "test");
+    ///     # #[allow(unused_variables)]
+    ///     let template = Template::show(client.rocket(), "index", context);
+    /// }
+    /// ```
     #[inline]
-    pub fn show<P, S, C>(root: P, name: S, context: C) -> Option<String>
-        where P: AsRef<Path>, S: Into<Cow<'static, str>>, C: Serialize
+    pub fn show<S, C>(rocket: &Rocket, name: S, context: C) -> Option<String>
+        where S: Into<Cow<'static, str>>, C: Serialize
     {
-        let root = root.as_ref().to_path_buf();
-        Context::initialize(root).and_then(|ctxt| {
-            Template::render(name, context).finalize(&ctxt).ok().map(|v| v.0)
-        })
+        let ctxt = match rocket.state::<Context>() {
+            Some(ctxt) => ctxt,
+            None => {
+                warn_!("Uninitialized template context: missing fairing.");
```

Comment:
Is this the correct indentation level? I think it should probably be `warn!` instead of `warn_!`, but best to trigger the warning manually and confirm with the log output.

Reply:
`warn_!` indents with `=>`; `warn!` does not indent. The indented versions seem to be more consistent with the rest of the crate and with `impl<Responder> for Template`, but to be honest I don't know which is more appropriate for this message.

Reply:
This should be `warn!` since it is being called from a top-level method.